pyscf/pbc/symm/group.py

```diff
@@ -64,244 +132,331 @@ def matrix(self, a):
+            _table = [self.hash_table[hash(gh)] for gh in prod.flatten()]
+            self._multiplication_table = np.asarray(_table).reshape(prod.shape)
+        return self._multiplication_table
+
+    @multiplication_table.setter
+    def multiplication_table(self, table):
+        self._multiplication_table = table
 
+    @property
     def conjugacy_table(self):
         '''
-        conjugacy_table[idx_g, idx_h] gives index of :math:`h^{-1} g h`.
+        conjugacy_table[`index_g`, `index_x`] returns the index of element `h`,
+        where :math:`h = x * g * x^{-1}`.
         '''
-        inv_table = self.inverse_table()
-        prod_table = self.multiplication_table()
-        ginv_h = prod_table[inv_table]
-        idx = np.arange(len(self))[None, :]
-        return ginv_h[ginv_h, idx]
+        if self._conjugacy_table is None:
+            prod_table = self.multiplication_table
+            g_xinv = prod_table[:,self.inverse_table]
+            self._conjugacy_table = prod_table[np.arange(self.order)[None,:], g_xinv]
+        return self._conjugacy_table
 
-    def conjugacy_classes(self):
+    @conjugacy_table.setter
+    def conjugacy_table(self, table):
+        self._conjugacy_table = table
+
+    @property
+    def conjugacy_mask(self):
         '''
-        Conjugacy classes. Unsorted
+        Boolean mask array indicating whether two elements
+        are conjugate with each other.
         '''
-        n = len(self)
-        idx = np.arange(len(self))[:, None]
-        is_conjugate = np.zeros((n,n), dtype=np.int32)
-        is_conjugate[idx, self.conjugacy_table()] = 1
+        if self._conjugacy_mask is None:
+            n = self.order
+            is_conjugate = np.zeros((n,n), dtype=bool)
+            is_conjugate[np.arange(n)[:,None], self.conjugacy_table] = True
+            self._conjugacy_mask = is_conjugate
+        return self._conjugacy_mask
 
-        classes, representatives, inverse = np.unique(is_conjugate, axis=0, return_index=True, return_inverse=True)
+    def conjugacy_classes(self):
+        '''
+        Compute conjugacy classes.
+
+        Returns:
+            classes : (n_irrep,n) boolean array
+                The indices of `True` correspond to the
+                indices of elements in this class.
+            representatives : (n_irrep,) array of ints
+                Representive elements' indices in each class.
+            inverse : (n,) array of ints
+                The indices to reconstruct `conjugacy_mask` from `classes`.
+        '''
+        _, idx = np.unique(self.conjugacy_mask, axis=0, return_index=True)
+        representatives = np.sort(idx)
+        classes = self.conjugacy_mask[representatives]
+        inverse = -np.ones((self.order), dtype=int)
+        diff = (self.conjugacy_mask[None,:,:]==classes[:,None,:]).all(axis=-1)
+        for i, a in enumerate(diff):
+            inverse[np.where(a==True)[0]] = i # noqa: E712
```

Comment:
why it's `np.where(a==True)`? Isn't it the same to `np.where(a)`?

Reply:
fixed